### PR TITLE
[CAFV-389] Add clusterclass support for v1beta3

### PR DESCRIPTION
**Pull request by Anirudh9794**
## Description
Please provide a brief description of the changes proposed in this Pull Request

- Adds v1beta3 types for VCDClusterTemplate API object
- Tested by creating a cluster using clusterclass

## Checklist
- [x] tested locally
- [x] updated any relevant dependencies
- [x] updated any relevant documentation or examples

## API Changes
Are there API changes?
- [x] Yes
- [ ] No

If yes, please fill in the below

1. Updated conversions?
    - [x] Yes
    - [ ] No
    - [ ] N/A
2. Updated CRDs?
    - [x] Yes
    - [ ] No
    - [ ] N/A
3. Updated infrastructure-components.yaml?
    - [x] Yes
    - [ ] No
    - [ ] N/A
4. Updated `./examples/capi-quickstart.yaml`?
    - [ ] Yes
    - [ ] No
    - [x] N/A - the sample file doesn't use clusterclass
5. Updated necessary files under `./infrastructure-vcd/v1.0.0/`?
   - [x] Yes
   - [ ] No
   - [ ] N/A

## Issue
If applicable, please reference the relevant issue

Fixes #

- - -
This change is [Reviewable](https://reviewable.io/reviews/vmware/cluster-api-provider-cloud-director/598)

### CODE REVIEW

**Review by arunmk — APPROVED** [review]
Can we get this reviewed by @viveksyngh and @jjaferson as well please

File: api/v1beta3/vcdclustertemplate_types.go [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// VCDClusterTemplateSpec defines the desired state of VCDClusterTemplate
+type VCDClusterTemplateSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// Foo is an example field of VCDClusterTemplate. Edit vcdclustertemplate_types.go to remove/update
+	Template VCDClusterTemplateResource `json:"foo,omitempty"`
```
Comment (viveksyngh) [inline api/v1beta3/vcdclustertemplate_types.go:32]: This should json key should be `template` and empty value should not be allowed.

File: api/v1beta3/vcdclustertemplate_conversion.go [hunk 2]
```diff
@@ -0,0 +1,5 @@
+package v1beta3
+
+func (*VCDClusterTemplate) Hub() {}
+
+func (*VCDClusterTemplateList) Hub() {}
```
Comment (viveksyngh) [inline api/v1beta3/vcdclustertemplate_conversion.go:5]: We need a conversion webhook to convert resources from v1beta2 to v1beta3 ?

Comment (Anirudh9794) [reply api/v1beta3/vcdclustertemplate_conversion.go:5]: A webhook for vcdclustertemplate is already present (under `config/crd/patches/webhook_in_vcdclustertemplates.yaml`)

File: api/v1beta2/vcdclustertemplate_conversion.go [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
+func Convert_v1beta3_VCDClusterTemplate_To_v1beta2_VCDClusterTemplate(in *v1beta3.VCDClusterTemplate, out *VCDClusterTemplate, s apiconversion.Scope) error {
+	return autoConvert_v1beta3_VCDClusterTemplate_To_v1beta2_VCDClusterTemplate(in, out, s)
+}
+
+func (src *VCDClusterTemplate) ConvertTo(dstRaw conversion.Hub) error {
```
Comment (viveksyngh) [inline api/v1beta2/vcdclustertemplate_conversion.go:14]: can we add comment for this public function ?

File: api/v1beta2/vcdclustertemplate_conversion.go [hunk 4]
```diff
@@ -0,0 +1,49 @@
+package v1beta2
+
+import (
+	"github.com/vmware/cluster-api-provider-cloud-director/api/v1beta3"
+	apiconversion "k8s.io/apimachinery/pkg/conversion"
+	utilconversion "sigs.k8s.io/cluster-api/util/conversion"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+)
+
```
Comment (viveksyngh) [inline api/v1beta2/vcdclustertemplate_conversion.go:9]: can we add comment for this public function ?

File: api/v1beta3/vcdclustertemplate_conversion.go (hunk 2 above)
Comment (viveksyngh) [reply api/v1beta3/vcdclustertemplate_conversion.go:5]: For some reason, I don't see this patch applied on the generated CRD

File: PROJECT [hunk 5]
```diff
@@ -145,4 +145,12 @@ resources:
   webhooks:
     conversion: true
     webhookVersion: v1
+- api:
```
Comment (viveksyngh) [inline PROJECT:148]: Here is we should webhooks section with conversion set to true ?

```
  webhooks:
    conversion: true
    webhookVersion: v1
```

File: api/v1beta3/vcdclustertemplate_conversion.go [hunk 6]
```diff
@@ -0,0 +1,5 @@
+package v1beta3
```
Comment (sahithi) [inline api/v1beta3/vcdclustertemplate_conversion.go:1]: Aniruddha, please update templates/cluster-class-template.yaml file (or) add a new file for v1beta3

**Review by sahithi — APPROVED** [review]
LGTM after tests are pass with VCDClusterTemplate.
Add an example file in the templates folder

**Comment by Anirudh9794** [comment]
Closing this PR and re-did the changes in #604

**Pull request closed without merging**